Tools/Scripts/compare-static-analysis-results

```diff
@@ -332,8 +341,35 @@ def main():
         if type_total:
             print(f'Total {type}: {len(type_total)}')
 
-    if args.report_urls:
-        upload_results(args, results_for_upload)
+    return results_for_upload
+
+
+def generate_results_only(args):
+    result_paths = []
+    print(f'Generating new results index with results from {args.build_output}...')
+
+    with open(os.path.abspath(f"{args.build_output}/unexpected_results.json"), "r") as f:
+        results_data = json.load(f)
+    with open(f'{args.new_dir}/issues_per_file.json') as f:
+        issues_per_file = json.load(f)
+
+    for project, checkers in results_data.get('failures').items():
+        for checker, files in checkers.items():
+            for file in files:
+                result_paths += [i for i in issues_per_file[checker][file].values()]
```

Comment:
issues_per_file.values() don't contain integers does it? It's confusing to call that variable "i".
If it's really an index, why is the aggregated result called "result_paths"?

Reply:
No, sorry, I used `i` as short for `issue`.

I think better naming is `report_paths`, since these are actually paths to the reports for each issue (e.g. `scan-build-output/StaticAnalyzer/WebCore/StaticAnalyzerReports/report-ContainerNode.cpp-parserInsertBefore-13-a66ea5.html`). We need to pass these into `create_filtered_results_dir` to create a filtered `StaticAnalyzerReports` folder that `scan-build --generate-index-only` uses.

I've made these changes, let me know what you think.